src/builder/XMLBuilderImpl.ts

```diff
@@ -168,7 +168,7 @@ export class XMLBuilderImpl implements XMLBuilder {
 
           // expand child nodes under parent
           lastChild.ele(val)
-        } else if (val) {
+        } else if (val != null && val !== '') {
```

Comment:
Can you please remove `val !== ''`. I don't see any reason to not allow empty text nodes.

Reply:
Sorry to be late.

When I was testing this script:

```js
const builder = require('xmlbuilder2');

[
  { headless: true, allowEmptyTags: false },
  { headless: true, allowEmptyTags: true },
].map(options => {
  const xml = builder
    .create()
    .ele({ value: '' })
    .end(options);
  console.log(options);
  console.log(xml);
});
```

I got this result using `val != null`:

```txt
{ headless: true, allowEmptyTags: false, format: 'xml' }
<value></value>
{ headless: true, allowEmptyTags: true, format: 'xml' }
<value></value>
```

And got this result using `val != null && val !== ''`:

```txt
{ headless: true, allowEmptyTags: false, format: 'xml' }
<value/>
{ headless: true, allowEmptyTags: true, format: 'xml' }
<value></value>
```

I thought the latter situation will meet my expectation.